### DISCUSSION TITLE
Syntax highlighting not working

### DISCUSSION THREAD

**Discussion by wiljam144** [nvim-neorg/neorg#162] [Q&A] [answered]
When i open .norg file syntax highlighting isn't working, I'm using vimscript to configure nvim. here is the lua part of the config:
``` lua
    local parser_configs = require('nvim-treesitter.parsers').get_parser_configs()

    parser_configs.norg = {
        install_info = {
            url = "https://github.com/nvim-neorg/tree-sitter-norg",
            files = { "src/parser.c", "src/scanner.cc" },
            branch = "main"
        },
    }

    require('nvim-treesitter.configs').setup {
	    ensure_installed = { "norg" },
    }

    require('neorg').setup {
        -- Tell Neorg what modules to load
        load = {
            ["core.defaults"] = {}, -- Load all the default modules
            ["core.norg.concealer"] = {}, -- Allows for use of icons
            ["core.norg.dirman"] = { -- Manage your directories with Neorg
                config = {
                    workspaces = {
                        my_workspace = "~/neorg"
                    }
                }
            },
        },
    }
```

**Comment by danymat** [comment]
Did you do `:TSInstall norg`?

**Reply by wiljam144** [reply]
yes

**Comment by mrossinek** [comment]
I think you will need to provide a bit more information than this. Here are some starter questions:

* what version of Neovim are you using?
* what version of Neorg are you using?
* are you using packer? If so, try disabling lazy loading (if you use that)
* check the loading order of nvim-treesitter and Neorg
* are any log messages being generated?
* does this happen consistently in all of the following cases?
    1. `nvim some_neorg_file.norg`
    2. `nvim` and then `:e some_neorg_file.norg`
    3. `nvim`, then `:Neorg` and then `:e some_neorg_file.norg`
    4. what about opening any of the help pages after starting nvim in any of the above cases? I.e. do one of the above and then `:help neorg`

**Reply by wiljam144** [reply]
neovim ver: 0.6.0-dev
neorg ver: latest
i'm not using packer
order is checked
no log messages
happens every time in all of cases you provided
links are highlighted but everything else not

**Reply by mrossinek** [reply]
That's very weird indeed :thinking: 

How are links highlighted?
Can you also install treesitter playground (https://github.com/nvim-treesitter/playground) so that we can check two things:
1. the output of `:TSPlaygroundToggle`
2. the output of `:TSHighlightCapturesUnderCursor` once while on a link and once while not

Also please provide the output of `:filter Neorg hi` (preferably as a screenshot to preserve the color information)

**Reply by wiljam144** [reply]
links highlighting:
![image](https://user-images.githubusercontent.com/58435806/138164853-486b6eed-e289-4f31-b4be-781e92a3bd71.png)
`:TSPlaygroundToggle` output:
![image](https://user-images.githubusercontent.com/58435806/138165109-7202255d-b1e8-4fd9-aef2-58a6706438d8.png)
when i try to execute `:TSHighlitCapturesUnderCursor` i get an error
![image](https://user-images.githubusercontent.com/58435806/138165626-3a1f6bef-3930-4f43-9a81-62d0377b90a9.png)
`:filter Neorg` output
![image](https://user-images.githubusercontent.com/58435806/138165750-b68ef16d-01a0-44bd-a4ca-1f6c93b4e80f.png)
![image](https://user-images.githubusercontent.com/58435806/138165781-80946f2b-5cbb-49fc-a650-9b06fd39ac34.png)
![image](https://user-images.githubusercontent.com/58435806/138165811-e4e2ec8d-998c-45ec-901d-e66664c81b6c.png)

**Comment by wiljam144** [comment, marked as answer]
I Fixed it, what i did was running `:TSEnableAll highlight` and then running `:e` and highlighting started working

**Reply by wiljam144** [reply]
to make vim do it automatically i added 
```lua
require'nvim-tresitter.configs'.setup {
    highlight = {
       enable = true
   }
}
```
into my config

**Reply by mrossinek** [reply]
Awesome! Thanks for letting us know! @vhyrro this is something we should add to the installation instructions! I can do that on the weekend :+1:

**Reply by vhyrro** [reply]
Oh lmao! We definitely need to add this to the `main` branch readme then
